### PR TITLE
Added qutip-qip tutorials to documentation on Read the docs

### LINKED ISSUE #204: Add qutip-qip tutorials to documentation on Read the docs
# Issue description
Add links for exisiting qutip-qip notebooks on the read the docs documentation at https://qutip-qip.readthedocs.io/en/stable/, similarly to what is done for Mitiq. Right now there is a tutorials section (see screenshot below) 

qutipqip-doc

but it only contains a link to qutip.org, as the tutorials appear only on the https://qutip.org/qutip-tutorials/ under the QIP section, see screenshot below.

qutip-qip

# Proposed Solution
The tricky part is that we don't want to duplicate the files, the source files should remain from `qutip-tutorials`, such as for the
"[Decomposition of the Toffoli gate in terms of CNOT and single-qubit rotations](https://nbviewer.org/urls/qutip.org/qutip-tutorials/tutorials-v4/quantum-circuits/qip-toffoli-cnot.ipynb)" that is rendered from
```
https://nbviewer.org/urls/qutip.org/qutip-tutorials/tutorials-v4/quantum-circuits/qip-toffoli-cnot.ipynb
```   
The tutorials should be kept also on the https://qutip.org/qutip-tutorials/#quantum-information-processing as they currently exist. 

# References
- https://qutip-qip.readthedocs.io/en/stable/
- https://qutip.org/qutip-tutorials/#quantum-information-processing
- https://github.com/qutip/qutip-tutorials

### PR DESCRIPTION
**Pull request by rum1887**
Closes: Add qutip-qip tutorials to documentation on Read the docs #204
Did not add the examples in the gallery form like in Mitiq. can be converted to that form easily tho 

Screenshot 2023-05-28 at 8 07 08 PM

Just added the links to the text

### CODE REVIEW

**Comment by rum1887** [comment]
hi , I have generated the html from qutip tutorials 
https://github.com/qutip/qutip-tutorials/pull/64

**Comment by rum1887** [comment]
> hi , I have generated the html from qutip tutorials
> [qutip/qutip-tutorials#64](https://github.com/qutip/qutip-tutorials/pull/64)

Planning on create a .py file to render the html to tutorials.rst

**Comment by rum1887** [comment]
> > hi , I have generated the html from qutip tutorials
> > [qutip/qutip-tutorials#64](https://github.com/qutip/qutip-tutorials/pull/64)
> 
> Planning on creating a python script to render the html to tutorials.rst

I have created a file called html2rst.py which renders the html into the .rst 

`import subprocess

def convert_html_to_rst(html_file_path, rst_file_path):
    # Use the subprocess module to call the pandoc command-line tool
    subprocess.run(['pandoc', html_file_path, '-o', rst_file_path])

html_file_path = 'qutip-qip.html'
rst_file_path = 'tutorials.rst'
convert_html_to_rst(html_file_path, rst_file_path)`

I'll have to provide the link to the html file generated previously for the qutip-qip tutorials.

I can locally generate the .html by running the script.

even the .html for the index file is not available in the github, when I run the create_index.py , I can locally generate the index.html

hope I communicated it well 
@BoxiLi how to add the html link

**Comment by rum1887** [comment]
> > > hi , I have generated the html from qutip tutorials
> > > [qutip/qutip-tutorials#64](https://github.com/qutip/qutip-tutorials/pull/64)
> > 
> > 
> > Planning on creating a python script to render the html to tutorials.rst
> 
> I have created a file called html2rst.py which renders the html into .rst
> 
> This is the html2rst.py file
> 
> ```python
> import subprocess
> 
> def convert_html_to_rst(html_file_path, rst_file_path):
>    # Use the subprocess module to call the pandoc command-line tool
>     subprocess.run(['pandoc', html_file_path, '-o', rst_file_path])
> 
> html_file_path = 'qutip-qip.html'
> rst_file_path = 'tutorials.rst'
> convert_html_to_rst(html_file_path, rst_file_path)
> ```
> 
> This is the cropped local version of the tutorials doc after rendering the html(need to remove the header , footer and title): Screenshot 2023-05-28 at 11 08 01 PM
> 
> I'll have to provide the link to the html file generated previously for the qutip-qip tutorials. [qutip/qutip-tutorials#64](https://github.com/qutip/qutip-tutorials/pull/64) file qutip-qip.html
> 
> I can locally generate the .html by running the script [create_qutip-qip.py](https://github.com/qutip/qutip-tutorials/pull/64/files#diff-0636c98068f0212c3cb40256cce8e1ab00f430f3671d89b4bb3a7ae468280f64) that I created by modifying the [create_index.py](https://github.com/qutip/qutip-tutorials/blob/main/website/create_index.py).
> 
> even the index.html for the index file is not available on the GitHub, when I run the create_index.py, I can locally generate the index.html
> 
> hope I communicated it well
> 
> @nathanshammah @BoxiLi how to add the HTML link, I can't find the index.html too on GitHub and I can't add a local link

TLDR:

- Have created the cropped version of HTML using the jinja template (html file is created when the script is run)
- A Python script to render the HTML into the tutorial.rst (need the deployed .html so I can use the link)

Issue: Need to get the link

**Comment by BoxiLi** [comment]
What you described sounds almost like a success.

I don't think we should push any code to `qutip-tutorial`. The files should be in qutip-qip.

The question is, do you really need the deployed HTML? If you can create it locally, you should be able to crop it from the file locally, or did I miss something?

I'm thinking more of something like this: The Python script is executed automatically when calling `make html`, which reads the source file from `qutip-tutorial`, generates the HTML locally, and then crops it to get the `rst` file. Do you think it is feasible?

**Comment by rum1887** [comment]
> What you described sounds almost like a success.
> 
> I don't think we should push any code to `qutip-tutorial`. The files should be in qutip-qip.
> 
> The question is, do you really need the deployed HTML? If you can create it locally, you should be able to crop it from the file locally, or did I miss something?
> 
> I'm thinking more of something like this: The Python script is executed automatically when calling `make html`, which reads the source file from `qutip-tutorial`, generates the HTML locally, and then crops it to get the `rst` file. Do you think it is feasible?

Alright, now I get it
I could locally create the HTML by running the file in the qutip-tutorials repository and not the qutip-qip repo.
I will try to figure out how to generate the HTML in the qutip-qip

**Comment by rum1887** [comment]
> What you described sounds almost like a success.
> 
> I don't think we should push any code to `qutip-tutorial`. The files should be in qutip-qip.
> 
> The question is, do you really need the deployed HTML? If you can create it locally, you should be able to crop it from the file locally, or did I miss something?
> 
> I'm thinking more of something like this: The Python script is executed automatically when calling `make html`, which reads the source file from `qutip-tutorial`, generates the HTML locally, and then crops it to get the `rst` file. Do you think it is feasible?

Feasible, doing it rn!

**Comment by rum1887** [comment]
will close this pr!
new pr reopened at #207
